From 40e87491886957696486b87dc2dedec2adaf6e1a Mon Sep 17 00:00:00 2001
From: Alexander Shevchenko <sav_ix@ukr.net>
Date: Wed, 10 Jan 2018 17:51:54 +0200
Subject: unify windows mkspecs: update definitions

mingw-w64 toolchain:
- add missing compiler definitions, similar to
  'msvc-desktop.conf' toolchain,
- describe the reasons of missing compiler definitions, available in
  'msvc-desktop.conf' toolchain,
- add missing 'QMAKE_CXXFLAGS' and 'QMAKE_CXXFLAGS_WARN_ON' variables,
  similar to 'msvc-desktop.conf' toolchain.

ICC on Windows toolchain:
- add 'QMAKE_CFLAGS_OPTIMIZE_FULL' variable, similar to 'gcc-base.conf'
  toolchain, though left it unused for now,
- add missing flags to 'QMAKE_CFLAGS' variable, similar to
  'msvc-desktop.conf' toolchain,
- update deprecated 'Qwd' flag with 'Qdiag-disable',
- use 'QMAKE_CFLAGS_OPTIMIZE_DEBUG' variable instead of '-Od' flag,
  similar to 'gcc-base.conf' toolchain (ICC implies '-O2' optimization
  level by default, while MSVC implies '-Od'),
- add 'QMAKE_CFLAGS_UTF8_SOURCE' variable, similar to
  'msvc-version.conf' toolchain; use a workaround to initialize it,
  until '-utf-8' flag would be supported by ICC on Windows,
- update deprecated '-Qstd=c++1z' flag with '-Qstd=c++17',

MSVC toolchain:
- remove 'incremental' from MSVC 'CONFIG' variable, since it has
  relevance only for the Unix generator,
- add 'QMAKE_CFLAGS_OPTIMIZE_DEBUG' variable, used in ICC for Windows
  toolchain,
- add empty 'QMAKE_LIBS' variable, similar to 'win32-g++' toolchain,
- add 'uuid.lib' library to 'QMAKE_LIBS_GUI' variable, similar to
  'win32-g++' toolchain,
- add C++14 and C++17 language support flags, though left them disabled
  for now, similar to 'win32-icc' toolchain.

Change-Id: Ideef62d0422674184836faa655bfc5d09a5f612f
Reviewed-by: Joerg Bornemann <joerg.bornemann@qt.io>
Reviewed-by: Kai Koehne <kai.koehne@qt.io>
Reviewed-by: Oswald Buddenhagen <oswald.buddenhagen@qt.io>
---
 mkspecs/common/msvc-desktop.conf | 6 ++++--
 mkspecs/common/msvc-version.conf | 5 +++++
 2 files changed, 9 insertions(+), 2 deletions(-)

(limited to 'mkspecs/common')

diff --git a/mkspecs/common/msvc-desktop.conf b/mkspecs/common/msvc-desktop.conf
index 496ca1179d..546b9cf3cb 100644
--- a/mkspecs/common/msvc-desktop.conf
+++ b/mkspecs/common/msvc-desktop.conf
@@ -15,7 +15,7 @@
 MAKEFILE_GENERATOR      = MSVC.NET
 QMAKE_PLATFORM          = win32
 QMAKE_COMPILER          = msvc
-CONFIG                 += incremental flat debug_and_release debug_and_release_target precompile_header autogen_precompile_source embed_manifest_dll embed_manifest_exe
+CONFIG                 += flat debug_and_release debug_and_release_target precompile_header autogen_precompile_source embed_manifest_dll embed_manifest_exe
 DEFINES                += UNICODE _UNICODE WIN32
 QMAKE_COMPILER_DEFINES += _WIN32
 contains(QMAKE_TARGET.arch, x86_64) {
@@ -23,6 +23,7 @@ contains(QMAKE_TARGET.arch, x86_64) {
     QMAKE_COMPILER_DEFINES += _WIN64
 }
 
+QMAKE_CFLAGS_OPTIMIZE_DEBUG = -Od
 QMAKE_CFLAGS_OPTIMIZE      = -O2
 QMAKE_CFLAGS_OPTIMIZE_SIZE = -O1
 
@@ -92,7 +93,8 @@ QMAKE_EXTENSION_SHLIB   = dll
 QMAKE_PREFIX_STATICLIB  =
 QMAKE_EXTENSION_STATICLIB = lib
 
-QMAKE_LIBS_GUI          = gdi32.lib comdlg32.lib oleaut32.lib imm32.lib winmm.lib ws2_32.lib ole32.lib user32.lib advapi32.lib
+QMAKE_LIBS              =
+QMAKE_LIBS_GUI          = gdi32.lib comdlg32.lib oleaut32.lib imm32.lib winmm.lib ws2_32.lib ole32.lib uuid.lib user32.lib advapi32.lib
 QMAKE_LIBS_NETWORK      = ws2_32.lib
 QMAKE_LIBS_OPENGL       = glu32.lib opengl32.lib gdi32.lib user32.lib
 QMAKE_LIBS_OPENGL_ES2   = gdi32.lib user32.lib
diff --git a/mkspecs/common/msvc-version.conf b/mkspecs/common/msvc-version.conf
index ba74c49f9c..cfafb06305 100644
--- a/mkspecs/common/msvc-version.conf
+++ b/mkspecs/common/msvc-version.conf
@@ -85,6 +85,11 @@ greaterThan(QMAKE_MSC_VER, 1909) {
     # API is used in direct2d, but also in multimedia, positioning and sensors.
     # We can try again with a later version of Visual Studio.
     # QMAKE_CXXFLAGS_STRICTCXX = -permissive-
+    # MSVC partially supports the following, but '__cplusplus' definition is set
+    # as for C++98 until MSVC fully conforms with C++14, see
+    # https://developercommunity.visualstudio.com/content/problem/139261/msvc-incorrectly-defines-cplusplus.html
+    # QMAKE_CXXFLAGS_CXX14 = -std:c++14
+    # QMAKE_CXXFLAGS_CXX1Z = -std:c++latest
 }
 
 greaterThan(QMAKE_MSC_VER, 1910) {
-- 
cgit v1.1-6-g87c4

